Sources/TuistGenerator/Mappers/ModuleMapMapper.swift

```diff
@@ -101,7 +101,14 @@ public final class ModuleMapMapper: WorkspaceMapping {
                     mappedSettingsDictionary[Self.otherCFlagsSetting] = updatedOtherCFlags
                 }
 
-                mappedTarget.settings = (mappedTarget.settings ?? .default).with(base: mappedSettingsDictionary)
+                let defaultSettings = mappedProject.settings.defaultSettings == .none ? .recommended :
+                mappedProject.settings.defaultSettings
+                let targetSettings = mappedTarget.settings ?? Settings(
+                    base: [:],
+                    configurations: [:],
+                    defaultSettings: defaultSettings
+                )
```

Comment:
Either I'm missing something, or this can be
```suggestion
                let targetSettings = mappedTarget.settings ?? Settings(
                    base: [:],
                    configurations: [:],
                    defaultSettings: mappedProject.settings.defaultSettings ?? .recommended
                )
```

Reply:
`.none` he is not an `Optional` case. So this syntax will not work here

Reply:
ah good point - I mistakenly thought it was optional, in that case we can fallback to the project's default settings, no need to further fallback to recommend if it's `none` to maintain the existing behaviour.

Thanks!

Reply:
Fixed!